java/src/main/java/vw/jni/NativeUtils.java

```diff
@@ -88,7 +88,7 @@ else if (osName.toLowerCase().contains("linux")) {
             if (version == null) {
                 throw new UnsupportedOperationException("Cannot determine linux version");
             }
-            return distro.trim().replaceAll(" ", "_") + "." + version;
+            return (distro.trim().replaceAll(" ", "_") + "_" + version).toLowerCase();
```

Comment:
Understand your concern about major/minor OS version. 
Using the `os-maven-plugin` based approach, one way around this would be to implement  https://github.com/trustin/os-maven-plugin/issues/15.
Another option would be to list all the native libraries included in the archive at runtime and then automatically pick the best match.
Not sure which would be the better approach.
With Linux, binary compatibility it's not so much about the distribution and version as much as it is about glibc and kernel version. In that respect, the latter approach could arguably be more robust.